custom_components/aseko_local/aseko_decoder.py

```diff
@@ -82,120 +104,161 @@ def _available_probes(
+    def _fill_salt_unit_data(unit: AsekoDevice, data: bytes) -> None:
         unit.salinity = data[20] / 10
         unit.electrolyzer_power = data[21] if data[29] & ELECTROLYZER_RUNNING else 0
         unit.electrolyzer_active = bool(data[29] & ELECTROLYZER_RUNNING)
         unit.electrolyzer_direction = AsekoDecoder._electrolyzer_direction(data)
 
     @staticmethod
-    def decode(data: bytes) -> AsekoDevice:
-        """Decode 120-byte array into AsekoData."""
+    def _fill_flowrate_data(unit: AsekoDevice, data: bytes) -> None:
+        def _normalize(value: int) -> int | None:
+            return None if value == 255 else value
 
+        unit.flowrate_chlor = _normalize(data[95])
+        unit.flowrate_ph_plus = _normalize(data[97])
+        unit.flowrate_ph_minus = _normalize(data[99])
+        unit.flowrate_floc = _normalize(data[101])
+
+    @staticmethod
+    def decode(data: bytes) -> AsekoDevice:
         unit_type = AsekoDecoder._unit_type(data)
         probes = AsekoDecoder._available_probes(data)
-        try:
-            timestamp = (
-                AsekoDecoder._timestamp(data)
-                if unit_type != AsekoDeviceType.NET
-                else datetime.now(tz=homeassistant.util.dt.get_default_time_zone())
-            )
-        except ValueError:
-            _LOGGER.warning(
-                "Invalid timestamp in the data: %s, using current timestamp instead",
-                data[6:12],
-            )
-            timestamp = datetime.now(tz=homeassistant.util.dt.get_default_time_zone())
+
+        ts = AsekoDecoder._timestamp(data)
+        _LOGGER.debug("Decoded timestamp = %s (raw: %s)", ts, data[6:12].hex())
+
+        # Pump type running (byte 29)
+        pump_code = int(data[29])
+        active_pump = (
+            AsekoPumpType(pump_code)
+            if pump_code in AsekoPumpType._value2member_map_
+            else AsekoPumpType.OFF
+        )
 
         device = AsekoDevice(
             serial_number=int.from_bytes(data[0:4], "big"),
-            type=unit_type,
-            timestamp=timestamp,
+            device_type=unit_type,
+            timestamp=AsekoDecoder._timestamp(data),
             water_temperature=int.from_bytes(data[25:27], "big") / 10,
             water_flow_to_probes=(data[28] == WATER_FLOW_TO_PROBES),
             pump_running=bool(data[29] & PUMP_RUNNING),
-            required_algicide=data[54],
-            required_water_temperature=data[55],
+            active_pump=active_pump,
+            required_algicide=AsekoDecoder._available_probes(data[54]),
+            required_water_temperature=AsekoDecoder._available_probes(data[55]),
             start1=AsekoDecoder._time(data[56:58]),
             stop1=AsekoDecoder._time(data[58:60]),
             start2=AsekoDecoder._time(data[60:62]),
             stop2=AsekoDecoder._time(data[62:64]),
-            backwash_every_n_days=data[68],
+            backwash_every_n_days=AsekoDecoder._available_probes(data[68]),
```

Comment:
When testing it fails for me on:
```
Traceback (most recent call last):
  File "/workspaces/aseko-local/custom_components/aseko_local/aseko_server.py", line 143, in _handle_client
    device = AsekoDecoder.decode(frame)
  File "/workspaces/aseko-local/custom_components/aseko_local/aseko_decoder.py", line 259, in decode
    backwash_every_n_days=AsekoDecoder._available_probes(data[68]),
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "/workspaces/aseko-local/custom_components/aseko_local/aseko_decoder.py", line 87, in _available_probes
    probe_info = data[4]
                 ~~~~^^^
TypeError: 'int' object is not subscriptable
```
Any particular reason not to use the byte `68` directly?

```suggestion
            backwash_every_n_days=data[68],
```